### DISCUSSION TITLE
Replacing the word Render with a more generic term ?

### DISCUSSION THREAD

**Discussion by nyue** [OpenJobDescription/openjd-specifications#24] [Q&A]
I am wondering if there is a more generic term to use instead of Render to signify OJD is more than just render management.

With the ongoing use of Machine Learning and AI workload in the creative industry, there might be opportunities to connect into other job management systems like [Slurm](https://www.schedmd.com/) and [PBS](https://en.wikipedia.org/wiki/Portable_Batch_System) either directly or indirectly via [CWL](https://en.wikipedia.org/wiki/Common_Workflow_Language).

**Comment by ddneilson** [comment]
Hi Nicholas. You're absolutely right. There's nothing about Open Job Description that ties it specifically to rendering. It's inspired by rendering workloads, but is certainly not only for it. At the end of the day, it's a mechanism for defining Jobs that consist of running command-line applications/scripts.

Some alternatives to "render job", off the top of my head:
* compute job
* headless batch command-line jobs
* batch command-line jobs

I'd love to hear other's opinions on this.

**Reply by jvanns** [reply]
I agree that keeping it general is the way to go -- especially as there is a plethora and decades of prior art here in the wider HPC (and operations research) communities that typically settle on the job -> tasks terms to describe workloads. In fact, generally, 'job' and 'workflow' are often used when describing a set of tasks that run arbitrary commands to achieve their goal. I think @shidarin raises a good point about maintaining the word 'render' in examples and such to keep things relevant and familiar. I'd shy away from using 'batch' though since its  a nuanced and overloaded term (a bit like 'render', I suppose!) not necessarily used to identify batch processing or batches any longer. Not in VFX/FA anyway.

'compute job' certainly makes sense to me though obviously not as cool as 'render' ;)

**Comment by pkpriority** [comment]
This is a really interesting topic. Nicholas I just want you to know I see this and I'm not ignoring this, but I mostly agree with Daniel and I'm leaving room for other opinions :) 

Naming aside, the Slurm, PBS, CWL note is really cool!

**Comment by shidarin** [comment]
You'll find a lot of agreement over here on that front. As I write and review material, I've been trying to interject 'compute' in areas where it makes sense to, while still allowing 'render' to be used for examples and descriptions, as that's where most of us cut our teeth and we find it still resonating with many of the folks we've been talking to.

It's very intentional that it's not called `Open Render Job Description`.

**Comment by pkpriority** [comment]
Last week at the ASWF Open Source Forum (2/22) when I talked about OpenJD, there were questions about using OpenJD for CICD purposes. I think that falls under the spirit of this thread.  Are there any CICD examples (... do people still use jenkins these days?) we can try writing up for the samples folder? General call for some non-[VFX|Anim|Render] commands to try turning into OpenJD templates
